app/controllers/api/v1/oidc/trusted_publisher_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Api::V1::OIDC::TrustedPublisherController < Api::BaseController
+  include ApiKeyable
+
+  before_action :decode_jwt
+  before_action :find_provider
+  before_action :verify_signature
+  before_action :find_trusted_publisher
+  before_action :validate_claims
+
+  class UnsupportedIssuer < StandardError; end
+  class UnverifiedJWT < StandardError; end
+
+  rescue_from(
+    UnsupportedIssuer, UnverifiedJWT,
+    JSON::JWT::VerificationFailed, JSON::JWK::Set::KidNotFound,
+    OIDC::AccessPolicy::AccessError,
+    with: :render_not_found
```

Comment:
404? These seem like format or auth problems not "we couldn't find the resource identified by that URI." Is there a behavior reason for this in github actions?

Reply:
given these are un-authed endpoints, we don't want to make it easy to snoop for configuration